packages/playground/src/components/ssh_keys/SshDataDialog.vue

```diff
@@ -95,16 +92,43 @@ export default defineComponent({
   setup(props, ctx) {
     const currentKey = ref<SSHKeyData>(props.selectedKey);
     const loading = ref<boolean>(false);
+    const hasChanges = ref<boolean>(true);
+    const originalKey = ref<SSHKeyData>({ ...props.selectedKey });
 
     watch(
       () => props.open,
       newValue => {
         if (newValue) {
           currentKey.value = { ...props.selectedKey };
+          originalKey.value = { ...props.selectedKey };
+          hasChanges.value =
+            currentKey.value.name !== originalKey.value.name ||
+            currentKey.value.fingerPrint !== originalKey.value.fingerPrint ||
+            currentKey.value.publicKey !== originalKey.value.publicKey;
```

Comment:
What is the idea of having those conditions in two different places?
After this watcher sets the current key with a new value, it should trigger the other watcher and update the hasChanges, right?